data/users/externalldap.py

```diff
@@ -217,7 +217,21 @@ def _ldap_user_search(self, username_or_email, limit=20, suffix=""):
             if err_msg is not None:
                 return (None, err_msg)
 
-            logger.debug("Found matching pairs: %s", pairs)
+            # This may reveal sensitive information. Hide it unless Quay is started
+            # with an explicit flag used for debugging user-related issues.
+            if os.environ.get("USERS_DEBUG") == "1":
```

Comment:
Do we ever need to log the resulting attributes? Just debug log the `DN` (first item in the pair) which should contain enough to debug on the ldap side if needed.

Reply:
Good suggestion. I've updated the code accordingly.